cclib/parser/gaussianparser.py

```diff
@@ -1853,46 +1857,86 @@ def natural_orbital_single_spin_parsing(inputfile, updateprogress_title):
+        #
+        # APT and Lowdin charges are also displayed in this way
+        def extract_charges_spins(prop,header):
+            phrases = [" atomic charges:",
+            " Atomic Charges:",
+            " charges:",
+            " charges with hydrogens summed into heavy atoms:",
+            " atomic charges with hydrogens summed into heavy atoms:",
+            " atomic spin densities:",
+            " charges and spin densities with hydrogens summed into heavy atoms:",
+            " charges and spin densities:"]
+            for phrase in phrases:
+                to_join = [prop,phrase]
+                if "".join(to_join) in line:
+        
+                    has_spin = 'spin' in line or 'Spin' in line
+                    has_charges = 'charges' in line or 'Charges' in line
+
+                    if has_charges and not hasattr(self, "atomcharges"):
+                        self.atomcharges = {}
+
+                    if has_spin and not hasattr(self, "atomspins"):
+                        self.atomspins = {}
+
+                    ones = next(inputfile)
+
+                    charges = []
+                    spins = []
+                    nline = next(inputfile)
 
-            elif "Lowdin" in line:
-                self.atomcharges["lowdin"] = charges
+                    # calculate how many lines need iterating over based on whether property is summed or not
+                    is_sum = 'summed' in line
+                    if is_sum:
+                        n = self.natom - self.nhydrogen
+                    else:
+                        n = self.natom
 
+                    # iterate over each line and append values to a list based on what property we have
+                    i = 0
+                    while i != n:
+                        if is_sum and nline.split()[1] == "H":  # some files included hydrogens in lists of summed charges with value 0.0, so these should not be recorded
+                            nline = next(inputfile)
+                        else:
+                            if has_charges:
+                                charges.append(float(nline.split()[2]))
+                            if has_spin and has_charges:
+                                spins.append(float(nline.split()[3]))
+                            if has_spin and not has_charges:
+                                spins.append(float(nline.split()[2]))
+                            if not i == n-1: # make sure we don't move to the header line for another set of values
+                                nline = next(inputfile)
+                            i +=1
+
+                    # input extracted values into self.atomcharges
+                    if prop in line:
+                        if has_charges:
+                            if is_sum:
+                                title_list = [header,"_charge_sum"]
+                                self.atomcharges["".join(title_list)] = charges
```

Comment:
again I would use f-strings here: remove title_list and:
`self.atomcharges[f"{header}_charge_sum"] = charges`

repeat below

Reply:
f strings are very useful!